pkg/parcacol/ingest.go

```diff
@@ -189,15 +189,14 @@ func validatePprofProfile(p *pprofproto.Profile) error {
 		if len(s.Value) != sampleLen {
 			return fmt.Errorf("mismatch: sample has %d values vs. %d types", len(s.Value), len(p.SampleType))
 		}
-		if len(s.LocationId) == 0 {
-			return fmt.Errorf("sample %d has no location ids", i)
-		}
-		for j, l := range s.LocationId {
-			if l == 0 {
-				return fmt.Errorf("location ids of stacktraces must be non-zero")
-			}
-			if l > locationsNum {
-				return fmt.Errorf("sample %d location number %d (%d) is out of range", i, j, l)
+		if len(s.LocationId) > 0 {
```

Comment:
This should be a no-op now, We can just remove the if as the for loop will only have an effect if there are location IDs.

Reply:
Doh :man_facepalming: